server/functions.lua

```diff
@@ -219,25 +219,33 @@ end
     ```
 ]]
 -- If you don't use the above on the client, it will return 0 as the vehicle from the netid and 0 means no vehicle found because it doesn't exist so fast on the client
+-- Deletes vehicle ped is in before spawning a new one.
 ---@param source number
 ---@param model string|number
 ---@param coords? vector4 default to player's position
 ---@param warp? boolean
 ---@return number? netId
 function QBCore.Functions.CreateVehicle(source, model, coords, warp)
+
     model = type(model) == 'string' and joaat(model) or model
     if not coords then coords = GetEntityCoords(GetPlayerPed(source)) end
     if not CreateVehicleServerSetter then
         error('^1CreateVehicleServerSetter is not available on your artifact, please use artifact 5904 or above to be able to use this^0')
         return
     end
+    local ped = GetPlayerPed(source)
+    local currentVeh = GetVehiclePedIsIn(ped, false)
+    if currentVeh and currentVeh ~= 0 then DeleteEntity(currentVeh) end
```

Comment:
Just a small thing, won't effect anything but you can remove the part where it checks if currentVeh is defined as GetVehiclePedIsIn never returns nil, it always returns a number